guava/src/com/google/common/math/Quantiles.java

```diff
@@ -551,18 +584,21 @@ private static void selectInPlace(int required, double[] array, int from, int to
    * equal to the value at {@code ret} and the values with indexes in ({@code ret}, {@code to}] are
    * greater than or equal to that.
    */
-  private static int partition(double[] array, int from, int to) {
+  @SuppressWarnings("lowerbound:compound.assignment.type.incompatible")// (1): Lowest possible `from` value is 0.
```

Comment:
I'd explain it like this:
Both `partitionPoint` and `i` are initialized to the same value (`to`).
`partitionPoint` is decremented at most one more time than `i` is decremented, therefore `partitionPoint >= i-1`.
`i > from`, therefore `partitionPoint >= from`. `from` is non-negative, therefore `partitionPoint` is non-negative.